swift/extractor/SwiftExtractor.cpp

```diff
@@ -176,10 +186,16 @@ void codeql::extractSwiftFiles(const SwiftExtractorConfiguration& config,
         continue;
       }
       archiveFile(config, *sourceFile);
-      extractDeclarations(config, compiler, *module, sourceFile);
+      encounteredModules = extractDeclarations(config, compiler, *module, sourceFile);
     }
     if (!isFromSourceFile) {
-      extractDeclarations(config, compiler, *module);
+      encounteredModules = extractDeclarations(config, compiler, *module);
+    }
+    processed.insert(module);
+    for (auto encountered : encounteredModules) {
+      if (processed.count(encountered) == 0) {
```

Comment:
This check should actually use seen modules, not already processed ones, or we can end up processing the same module multiple times. I will fix this tomorrow.